manager/secret_manager.go

```diff
@@ -21,14 +27,30 @@ func (r *ResourceManager) CreateSecret(namespace string, secret *model.Secret) (
 func (r *ResourceManager) SecretExists(namespace string, name string) (exists bool, err error) {
 	var foundSecret *model.Secret
 	var statusError *errors.StatusError
+
 	foundSecret, err = r.kubeClient.SecretExists(namespace, name)
 	if err != nil {
 		if goerrors.As(err, &statusError) {
 			if statusError.ErrStatus.Reason == "NotFound" {
+				logging.Logger.Log.WithFields(log.Fields{
+					"Namespace": namespace,
+					"Name":      name,
+					"Error":     err.Error(),
+				}).Warning("Secret not found.")
 				return false, util.NewUserError(codes.NotFound, "Secret Not Found.")
 			}
+			logging.Logger.Log.WithFields(log.Fields{
+				"Namespace": namespace,
+				"Name":      name,
+				"Error":     err.Error(),
+			}).Error("Checking existence of secret.")
 			return false, util.NewUserError(codes.Unknown, "Error when checking existence of secret.")
 		}
+		logging.Logger.Log.WithFields(log.Fields{
```

Comment:
So basically move this up right under `if err !== nil` and log `err` once. Thoughts?

Reply:
Hmm, I think I see your point. Moving it up right under err !== nil, would capture the error.
The rest if-else checks are for user reporting.

Should this adjustment be applied elsewhere too?

Reply:
That's correct. Yes, we should adjust this everywhere else.

Reply:
Right on, will do